src/lib/components/Cytoscape.react.js

```diff
@@ -196,6 +202,94 @@ class Cytoscape extends Component {
         }, SELECT_THRESHOLD);
 
         // /////////////////////////////////////// EVENTS //////////////////////////////////////////
+        const updateContextMenuData = (newContext) => {
+            this.props.setProps({contextMenuData: newContext});
+        };
+        const contextMenuDefaultFunctions = {
+            remove: function (event) {
+                let target = event.target || event.cyTarget;
+                target.remove();
+            },
+            add_node: function (event) {
+                let pos = event.position || event.cyPosition;
+                cy.add({
+                    data: {
+                        group: 'nodes',
+                    },
+                    position: {
+                        x: pos.x,
+                        y: pos.y,
+                    },
+                });
+            },
+            add_edge: function (event) {
+                const selectedNodeIds = selectedNodes.map((node) => node.id());
+                if (selectedNodes.length === 1) {
+                    cy.add({
+                        data: {
+                            id: uuidv4(),
+                            group: 'edges',
+                            source: selectedNodeIds[0],
+                            target: selectedNodeIds[0],
+                        },
+                    });
+                } else if (selectedNodes.length === 2) {
+                    cy.add({
+                        data: {
+                            id: uuidv4(),
+                            group: 'edges',
+                            source: selectedNodeIds[0],
+                            target: selectedNodeIds[1],
+                        },
+                    });
+                } else {
+                    console.log(
+                        'Error: more than 2 nodes selected, cannot add edge'
+                    );
+                }
+            },
+        };
+
+        const createMenuItems = (ctxMenu) => {
+            const new_menu_items = [];
+            for (const item of ctxMenu) {
+                let onClickFunction;
+                // use default javascript function
+                if (contextMenuDefaultFunctions.hasOwnProperty(item.onClick)) {
+                    onClickFunction = contextMenuDefaultFunctions[item.onClick];
+                }
```

Comment:
Should probably print something to the logs in the `else` case (to warn the developer that they've specified a built-in function which doesn't exist)